public/app/features/serviceaccounts/ServiceAccountCreatePage.tsx

```diff
@@ -75,7 +82,11 @@ export const ServiceAccountCreatePage = ({}: Props): JSX.Element => {
           tokens: response.tokens,
         };
         await updateServiceAccount(response.id, data);
-        if (contextSrv.licensedAccessControlEnabled()) {
+        if (
+          contextSrv.licensedAccessControlEnabled() &&
+          contextSrv.hasPermission(AccessControlAction.ActionUserRolesAdd) &&
+          contextSrv.hasPermission(AccessControlAction.ActionUserRolesRemove)
+        ) {
```

Comment:
note: this is to prevent the frontend from calling `updateUserRoles` if the user doesn't have the right permissions. To be honest, I don't really understand why it's being called here on the first place.